src/main/java/org/terasology/module/inventory/events/RequestInventoryEvent.java

```diff
@@ -0,0 +1,10 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+package org.terasology.module.inventory.events;
+
+import org.terasology.engine.entitySystem.event.Event;
+
+/**
+ */
+public class RequestInventoryEvent implements Event {
```

Comment:
As explained below, I suggest to make this event contain the payload of what items to add, similar to how the `StartingInventoryComponent` looks like. You'd likely need to move the nested class `InventoryItem` out of the component for this.

We could later extend this event with more properties, e.g., a boolean flag to determine whether the items should add to or override the inventory.

Reply:
The existing event handler for the starting inventory should also send an instance of this event, filled from the `StartingInventoryComponent`